### DISCUSSION TITLE
Custom MIDI

### DISCUSSION THREAD

**Discussion by reynmusician** [KBLiveSolutions/open.control#21] [Feature Requests & Ideas]
Hi Pierre!

Thanks for making a lovely tool!

I noticed that when you assign a Custom MIDI message it's only possible to do so from MIDI channel 1 to 10. I was wondering if it's possible to extend that to all 16 channels as I've already have loads of pre-set messages in my Ableton live-set.

Cheers, Reyn

Instagram.com/reynmusician
Facebook.com/reynmusician
YouTube.com/reynmusician

**Comment by KBLiveSolutions** [comment]
Hi Reyn

It is possible to access the extra channels by modifying the code slightly.
You just have to be aware that it may cause conflicts with the Notes that are assigned when using  Looper MIDI Map actions (which use Channels 11 to 16), and you may see some weird actions selected in the menus.
But most of the time it should be fine.

To access the extra channels, you need to [download the controller editor](https://github.com/KBLiveSolutions/open.control-editor/archive/refs/heads/main.zip).
Then, open it in a text editor and find the following text (line 556) : 
`var ch_values_custom = range(10, 1);`

you just need to change the `10` to `16`, save the file and open it in Chrome.
The custom MIDI will now show the 16 channels.

Cheers
p

**Comment by reynmusician** [comment]
Totally worked! Thanks Pierre.
